### PR TITLE
Preview live frame

### PR DESCRIPTION
**Pull request by zolga**
Adding a preview of the embedded link,  and copy link to the clipboard of the link.

image

### CODE REVIEW

File: src/components/PreviewFrame.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  return (
+    <Container>
+      <Row className='d-flex justify-content-center align-items-center p-1'>
+        <Col fluid='true'>
+          <iframe
+            src={'/live'}
+            width='400px'
+            height='500px'
+            frameBorder='0'
+            title='my-frame'
```
Comment (rylandg) [inline src/components/PreviewFrame.js]: nit: this title is pretty uninformative.

File: src/components/Admin/Admin.js [hunk 2]
```diff
@@ -65,41 +68,48 @@ export default function Admin() {
...
+              </Col>
+            </Row>
+          ))}
         </Col>
-        <Col className='col-1 pl-1 pr-0'>
-          <Button onClick={handleAddLink} className='url-add'>
-            +
-          </Button>
+        <Col>
+          <PreviewFrame key={frameKey} />
```
Comment (rylandg) [inline src/components/Admin/Admin.js]: PreviewFrame doesn't actually take in a key.

File: src/components/InputCopy.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      <InputGroup.Append>
+        <OverlayTrigger
+          key='bottom'
+          placement='bottom'
+          trigger='click'
+          overlay={
+            <Tooltip id={`tooltip-bottom`}>Copied to clipboard! </Tooltip>
+          }
+        >
+          <Button variant='outline-secondary' className='' onClick={handleCopy}>
```
Comment (rylandg) [inline src/components/InputCopy.js]: Why is there a classname ''

File: src/components/InputCopy.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        aria-label={url}
+        aria-describedby='basic-addon2'
+      />
+      <InputGroup.Append>
+        <OverlayTrigger
+          key='bottom'
+          placement='bottom'
+          trigger='click'
+          overlay={
+            <Tooltip id={`tooltip-bottom`}>Copied to clipboard! </Tooltip>
```
Comment (rylandg) [inline src/components/InputCopy.js]: the tooltip id could be using regular quotes ''

Comment (rylandg) [reply src/components/InputCopy.js]: Also no JS brackets {} are needed

File: src/components/PreviewFrame.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import Col from 'react-bootstrap/Col';
+import InputCopy from './InputCopy';
+
+export default function PreviewFrame() {
+  return (
+    <Container>
+      <Row className='d-flex justify-content-center align-items-center p-1'>
+        <Col fluid='true'>
+          <iframe
+            src={'/live'}
```
Comment (rylandg) [inline src/components/PreviewFrame.js]: No braces needed {}

File: src/components/InputCopy.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import InputGroup from 'react-bootstrap/InputGroup';
+import FormControl from 'react-bootstrap/FormControl';
+import Button from 'react-bootstrap/Button';
+import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
+import Tooltip from 'react-bootstrap/Tooltip';
+import copy from 'clipboard-copy';
+
+export default function InputCopy() {
+  const [action, setAction] = useState('COPY');
+  const url = window.location.href + 'live';
```
Comment (rylandg) [inline src/components/InputCopy.js:13]: nit: probably should use template string instead of `+`

File: src/components/Admin/Admin.js [hunk 7]
```diff
@@ -65,41 +68,48 @@ export default function Admin() {
...
+              </Button>
+            </Col>
+          </Row>
+
+          {linksList.map(url => (
+            <Row className='ml-0 url-row' key={url} variant='info'>
+              <Button
+                variant='light'
+                size='sm'
+                className=''
```
Comment (rylandg) [inline src/components/Admin/Admin.js]: empty className

File: src/components/Admin/Admin.js [hunk 8]
```diff
@@ -65,41 +68,48 @@ export default function Admin() {
...
+                onClick={() => handleDeleteList(url)}
+              >
+                X
+              </Button>
+              <Col className='col-10 trim-text '>
```
Comment (rylandg) [inline src/components/Admin/Admin.js]: nit: extra space

File: src/components/Admin/Admin.js [hunk 9]
```diff
@@ -65,41 +68,48 @@ export default function Admin() {
...
+                <span className=''>{url}</span>
```
Comment (rylandg) [inline src/components/Admin/Admin.js]: empty className

**Review by rylandg — CHANGES_REQUESTED** [review]
For the most part it's a good change. I think PreviewFrame needs to be updated to take in a key, also can you help me understand the purpose of `frameKey`?

File: src/components/Admin/Admin.js (hunk 7 above)
Comment (zolga) [reply src/components/Admin/Admin.js]: Removed.

File: src/components/Admin/Admin.js (hunk 8 above)
Comment (zolga) [reply src/components/Admin/Admin.js]: Removed

File: src/components/Admin/Admin.js (hunk 9 above)
Comment (zolga) [reply src/components/Admin/Admin.js]: Removed

File: src/components/InputCopy.js (hunk 6 above)
Comment (zolga) [reply src/components/InputCopy.js:13]: Changed.

File: src/components/InputCopy.js (hunk 4 above)
Comment (zolga) [reply src/components/InputCopy.js]: Changed.

File: src/components/InputCopy.js (hunk 3 above)
Comment (zolga) [reply src/components/InputCopy.js]: Removed

File: src/components/PreviewFrame.js (hunk 5 above)
Comment (zolga) [reply src/components/PreviewFrame.js]: Removed, thanks

File: src/components/PreviewFrame.js (hunk 1 above)
Comment (zolga) [reply src/components/PreviewFrame.js]: Changed

File: src/components/Admin/Admin.js (hunk 2 above)
Comment (zolga) [reply src/components/Admin/Admin.js]: Removed, unused code.

**Review by zolga — COMMENTED** [review]
Thanks! Changed and removed by your suggestions.
@rylandg  PTAL

File: src/components/Admin/Admin.js [hunk 10]
```diff
@@ -15,15 +15,14 @@ import './Admin.css';
...
-  const [frameKey, setFrameKey] = useState(1);
 
-  useEffect(async () => {
+  useEffect(() => {
     async function fetchFromDb() {
       const links = await getLinks();
       setLinksList(links);
     }
     try {
-      await fetchFromDb();
```
Comment (rylandg) [inline src/components/Admin/Admin.js:24]: This `try/catch` serves no purpose.

File: src/components/LiveFrame.js [hunk 11]
```diff
@@ -14,15 +14,15 @@ export default function Admin() {
...
 
-  useEffect(async () => {
+  useEffect(() => {
     async function fetchFromDb() {
       const links = await getLinks();
       setLinksList(links);
       const random = generateRandomListItem(links);
       setUrl(links[random]);
     }
     try {
```
Comment (rylandg) [inline src/components/LiveFrame.js]: Same as above, no point in this try.

**Review by rylandg — CHANGES_REQUESTED** [review]
I would prefer that you remove the now unused `try/catch`. Will approve after that!

Thanks for the quick fixes.

**Comment by zolga** [comment]
@rylandg you right, removed, thanks!

**Review by rylandg — APPROVED** [review]
The errors aren't handled but otherwise good!

**Pull request merged**